contracts/core/src/contract.rs

```diff
@@ -371,6 +375,29 @@ fn execute_set_non_native_rewards_receivers(
     ))
 }
 
+fn execute_update_withdrawn_amount(
+    deps: DepsMut<NeutronQuery>,
+    _env: Env,
+    info: MessageInfo,
+    batch_id: u128,
+    withdrawn_amount: Uint128,
+) -> ContractResult<Response<NeutronMsg>> {
+    let config = CONFIG.load(deps.storage)?;
+    if info.sender != config.withdrawal_manager_contract {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let mut batch = unbond_batches_map().load(deps.storage, batch_id)?;
```

Comment:
just in case, let's check wether batch status is withdrawn

Reply:
+1, will approve once this is resolved